src/platform/graphics/linux_dmabuf.cpp

```diff
@@ -101,7 +101,18 @@ class mg::DmaBufFormatDescriptors
                     nullptr,
                     &num_modifiers) != EGL_TRUE)
             {
-                BOOST_THROW_EXCEPTION((mg::egl_error("Failed to query number of modifiers")));
+                mir::log_warning("eglQueryDmaBufModifiers failed for format %s: %s",
+                    mg::drm_format_to_string(static_cast<uint32_t>(format)),
+                    mg::egl_category().message(eglGetError()).c_str());
+
+                // Remove that format and its modifiers from our list
+                formats.erase(formats.begin() + i);
+                modifiers_for_format.erase(modifiers_for_format.begin() + i);
+                external_only_for_format.erase(external_only_for_format.begin() + i);
+                // formats[i] is now the format *after* the one we've just removed, so go back one
+                // and continue the loop.
+                --i;
```

Comment:
It is a bit fragile and ugly relying on underflow being defined for unsigned types. I think I'd go for an empty _iteration-expression_ and an increment at the end of the loop.

Reply:
Yeah, fair enough. Done!